Add missing blocks from 1.10/1.11/1.12

Gloorf opened the pull request:
Also removes duplicates & trailing whitespace in the blocks definitions

TonyGravagno commented:
I'd take it but @walter is strongly adverse to whitespace changes, so you'll have to wait for his decision on a few of the lines.

Gloorf commented:
I'm just removing a couple of trailing whitespaces (because usually there is none in the file), i'm not changing the whole indentation scheme of the document :)

walterhiggins commented:
Hi @Gloorf 
If you can reinstate _stone_ and _fence_ I'm happy to merge this.

Gloorf commented:
stone (actually it's slab.stone) is already defined line 98 : https://github.com/walterhiggins/ScriptCraft/blob/development/src/main/js/modules/blocks.js#L98 
and fence is actually overwritten by the fence object on line 323 : https://github.com/walterhiggins/ScriptCraft/blob/development/src/main/js/modules/blocks.js#L323 so it was never used anyway. I fixed the fence object itself (correct data value for oak, added dark oak & nether brick fence)

jonathanperret commented:
Looks fine to me, although I'd have picked `fence.netherbrick` over `fence.nether_brick` since we already have `slab.netherbrick`.
